src/uwtools/config/formats/base.py

```diff
@@ -35,7 +35,7 @@ def __init__(self, config: Union[dict, OptionalPath] = None) -> None:
             self.update(config)
         else:
             self._config_file = str(config) if config else None
-            self.update(self._load(self._config_file))
+            self.data = self._load(self._config_file)
```

Comment:
`NMLConfig`'s `_load()` method now returns a `Namelist` object instead of a `dict`. Since `Namelist` objects conform to the `dict` interface, replacing the `data` attribute (a `dict`, per `UserDict` > `Config` > `NMLConfig` inheritance) with one should be correct. The `_load()` methods of the other `Config` children return `dict` objects.